### PR TITLE
Milestone 2 : Create add remove functionality

### PR DESCRIPTION
**Pull request by zdnahom**
## Milestone 2: Create add remove functionality
In this milestone I : 
- created new module files to divide my code
- implemented add todo function
- implemented remove todo function
- implemented edit todo function
- used localstorage to store my todos
- implemented each UI interaction like when clicking the remove button icon it immediately removes the todo from the UI and when you add a new to-do the UI will be updated. And also you can edit your to-do and press enter or go to another to-do list to save your update.

### CODE REVIEW

File: src/modules/TodosContainer.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  };
+
+  removeTodo = (id) => {
+    this.todos = this.todos
+      .filter((todo) => todo.id !== id)
+      .map((todo, index) => ({ ...todo, index: index + 1 }));
+    setTodos(this.todos);
+    UI.remove(id);
+  };
+}
```
Comment (VICTOR-CHUKWU) [inline src/modules/TodosContainer.js]: - [ ] kindly add method to edit todo as it is missing

File: src/modules/TodosContainer.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  todos = getTodos() || [];
+
+  addTodo = (description) => {
+    const id = (Math.floor(Math.random() * 100000)).toString();
+    const index = this.todos.length + 1;
+    const todo = new Todo(id, description, false, index);
+    this.todos.push(todo);
+    setTodos(this.todos);
+    UI.add(id, description);
+  };
```
Comment (VICTOR-CHUKWU) [inline src/modules/TodosContainer.js]: - [ ] I could add an empty string as todo, kindly add a check for that

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @zdnahom  👋  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit
- All linters passed


## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/TodosContainer.js (hunk 1 above)
Comment (zdnahom) [reply src/modules/TodosContainer.js]: @VICTOR-CHUKWU I heard from someone that it's optional.

File: src/modules/TodosContainer.js (hunk 2 above)
Comment (zdnahom) [reply src/modules/TodosContainer.js]: Thanks for the review, I will fix that.

File: src/modules/TodosContainer.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import Todo from './Todo.js';
+import { getTodos, setTodos } from './store.js';
+import UI from './UI.js';
+
+export default class TodosContainer {
+  todos = getTodos() || [];
+
```
Comment (MahmoudBakr23) [inline src/modules/TodosContainer.js:7]: - [ ] As reqested by the previous reviewer and the [project requirements](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md#:~:text=Implement%20a%20function%20for%20editing%20task%20descriptions.), please implement the editing function. 🙏 

_Thank you for communicating with the reviewers ❤️ However, please keep the project requirement as the first resource to get back to when you're confused between the required and the optional tasks._

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hi @zdnahom 😃, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

> Keep in mind that your long hours of work are appreciated. However, the code review process is for improving the code you wrote, and not for criticizing it. 😄

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

N/A

Cheers and Happy coding!👏👨‍💻💻

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/TodosContainer.js (hunk 1 above)
Comment (VICTOR-CHUKWU) [reply src/modules/TodosContainer.js]: @zdnahom please it is not optional

Comment (zdnahom) [reply src/modules/TodosContainer.js]: @VICTOR-CHUKWU You are right I'm very sorry 🙏 , I didn't clearly check the requirement.I will Implement it now 👍 .

File: src/modules/TodosContainer.js (hunk 3 above)
Comment (zdnahom) [reply src/modules/TodosContainer.js:7]: @MahmoudBakr23 I will do that, thanks for the review 🙏 .

**Review by Wes-Isaac — APPROVED** [review]
Hi @zdnahom, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- All functionalities are implemented 👍.
- Professional README 💯.
- Great communication with reviewers 🥇.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**